.github/workflows/pytest.yml

```diff
@@ -0,0 +1,45 @@
+name: pytest (RHEL 8, RHEL 9)
+
+on:
+  pull_request:
+    paths:
+      - "src/*"
+
+jobs:
+  pytest:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: "Fedora Rawhide"
+            image: "registry.fedoraproject.org/fedora:rawhide"
+          - name: "Fedora Latest"
+            image: "registry.fedoraproject.org/fedora:latest"
+          - name: "CentOS Stream 9"
+            image: "quay.io/centos/centos:stream9"
+          - name: "CentOS Stream 8"
+            image: "quay.io/centos/centos:stream8"
+
+    runs-on: "ubuntu-latest"
+    container:
+      image: ${{ matrix.image }}
+
+    steps:
+      - name: "Checkout the repository"
+        uses: actions/checkout@v4
+
+      - name: "Install dependencies"
+        # Setuptools is required because we still rely on 'distutils' which have been dropped in Python 3.12
+        run: |
+          dnf --setopt install_weak_deps=False install -y git-core python3-pip
+          python3 -m pip install --upgrade pip wheel
+          python3 -m pip install -r src/insights_client/tests/requirements.txt
+          python3 -m pip install git+https://github.com/RedHatInsights/insights-core.git
```

Comment:
I think it'd be better to use the egg that is already in the sources (see `data/rpm.egg`), pointing the tests to it via the `EGG` environment variable

you can also write a `conftest.py` for the tests that would set the `EGG` environment variable if not already set; this way, you can run the test suite with a custom egg, if needed/wanted

Reply:
a potential `src/insights_client/tests/conftest.py` might be:
```python
def pytest_configure(config):
    top = pathlib.Path(__file__).parent.parent.parent.parent
    if "EGG" not in os.environ:
        os.environ["EGG"] = str((top / "data" / "rpm.egg").resolve())
```